audinterface/core/segment.py

```diff
@@ -533,7 +535,21 @@ def process_signal_from_index(
             progress_bar=self.process.verbose,
             task_description=f'Process {len(index)} segments',
         )
-        index = audformat.utils.union(y)
```

Comment:
I was surprised to see that we were using `audformat.utils.union()` here also for the case with a file level. I was not expecting this could work...

Reply:
Ah right, forgot we changed it to support all indices. This will make it difficult to optimize in a similar way as we do here.